core/src/ast_builder/expr/mod.rs

```diff
@@ -192,12 +197,14 @@ pub fn expr(value: &str) -> ExprNode {
 }
 
 pub fn col(value: &str) -> ExprNode {
-    let idents = value.split('.').collect::<Vec<_>>();
+    let idents = value.split('.').map(|v| v.to_string()).collect::<Vec<_>>();
 
-    if idents.len() == 1 {
-        ExprNode::Identifier(value.to_owned())
-    } else {
-        ExprNode::CompoundIdentifier(idents.into_iter().map(ToOwned::to_owned).collect())
+    match &idents[..] {
+        [alias, ident] => ExprNode::CompoundIdentifier {
+            alias: alias.to_owned(),
+            ident: ident.to_owned(),
+        },
+        _ => ExprNode::Identifier(value.to_owned()),
```

Comment:
I'm sure there's better code... I'll think about it.

Reply:
The problem here is that `fn col()` is trying to handle both of `CompoundIdentifier` and `Identifier`.
Do you think it should throw error if `idents.len()` is not 2 or 1 ?

Reply:
Well, I think so.
It seems good to state that we do not support anything other than what we currently support.

Reply:
Though it changes some existing behavior but I think that is acceptable and does not have issue for general use cases.

here is just nit codes suggestion
It looks that `map(|v| v.to_string())` is unnecessary.

```rust
    let idents = value.split('.').collect::<Vec<_>>();

    match idents.as_slice() {
        [alias, ident] => ExprNode::CompoundIdentifier {
            alias: alias.to_string(),
            ident: ident.to_string(),
        },
        _ => ExprNode::Identifier(value.to_owned()),
    }
```